test/test_binary_ufuncs.py

```diff
@@ -1326,27 +1327,136 @@ def test_rdiv(self, device, dtype):
         z = torch.tensor([30 / v.item() for v in x], device=device)
         self.assertEqual(y, z, exact_dtype=False)
 
-    @onlyCPU
+    @dtypes(*torch.testing.get_all_fp_dtypes(include_bfloat16=False))
+    def test_fmod_by_zero_float(self, device, dtype):
+        # check floating-point tensor fmod to zero is nan on both CPU and GPU
+        x = make_tensor((10, 10), device=device, dtype=dtype, low=-9, high=9)
+        zero = torch.zeros_like(x)
+
+        self.assertTrue(torch.all(x.fmod(0.0).isnan()))
+        self.assertTrue(torch.all(x.fmod(zero).isnan()))
+        # out
+        out = torch.empty(0, device=device, dtype=dtype)
+        torch.fmod(x, zero, out=out)
+        self.assertEqual(out.size(), torch.Size([10, 10]))
+        self.assertTrue(torch.all(out.isnan()))
+        # in-place
+        x.fmod_(zero)
+        self.assertTrue(torch.all(x.isnan()))
+
+    @onlyOnCPUAndCUDA  # Check Issue https://github.com/pytorch/pytorch/issues/48130
```

Comment:
I guess this is undefined behavior lol

Reply:
Yes. And I currently don't have a plan to add tests regarding these corner cases on ROCM and XLA, because it would be just different value (`inf` or something else) and there is no document for reference.